civis/resources/_resources.py

```diff
@@ -166,14 +169,20 @@ def create_signature(args, kwargs):
         a dynamically created function.
     """
     p = [Parameter(x, Parameter.POSITIONAL_OR_KEYWORD) for x in args]
+    if six.PY3:
+        opt_par_type = Parameter.KEYWORD_ONLY
```

Comment:
Why `KEYWORD_ONLY` here?  IIUC, this would mean a function like this

```python
def f(a, b, c=3):
    return a + b + c
```

Could not be called like `f(1, 2, 3)` and would have to be called with `f(1, 2, c = 3)`.  Granted, since we've been using kwargs up until now, this won't break any code to use `KEYWORD_ONLY`.  But I think most users would expect the `POSTITIONAL_OR_KEYWORD` calling convention.

Reply:
If you're defining the function yourself, it would be
```python
def f(a, b, *, c=3):
    return a + b + c
```
and yes, you couldn't call it as `f(1, 1, 1)`.

The `KEYWORD_ONLY` preserves the existing behavior. Is the order of the optional arguments guaranteed to be stable? New arguments get added to API endpoints sometimes. Are new arguments guaranteed to always come after any existing arguments? If so, it seems okay to use `POSITIONAL_OR_KEYWORD`. However, I think that enforcing keyword only seems closer to the way the API is used, and more future-proof.

Reply:
Yeah, I agree that it is a cleaner API to have `KEYWORD_ONLY` given that we don't have guarantees that the order of keyword args is not going to change.  I do think this will confuse some users, but I suppose it would be better to give a confusing error during development than have code break later because argument order changed.